src/Microsoft.Dnx.Host.Mono/EntryPoint.cs

```diff
@@ -205,4 +220,83 @@ private static int BootstrapperOptionValueNum(string candidate)
         // It isn't a bootstrapper option
         return -1;
     }
+
+    private static void GetOsDetails(out string operatingSystem, out string osVersion, out string architecture)
+    {
+        var unameOutput = RunProgram("uname", "-s -m").Split(' ');
+
+        operatingSystem = unameOutput[0];
+        architecture = TranslateArchitecture(unameOutput[1]);
+        if (operatingSystem == "Darwin")
+        {
+            // sw_vers returns versions in format "10.10.4" so ".4" needs to be removed
+            osVersion = RunProgram("sw_vers", "-productVersion");
+            var lastDot = osVersion.LastIndexOf('.');
+            osVersion = lastDot >= 0 ? osVersion.Substring(0, lastDot - 1) : osVersion;
+            return;
+        }
+
+        try
+        {
+            var lsb_release = RunProgram("lsb_release", "-s -i -r");
+            if (!string.IsNullOrEmpty(lsb_release))
+            {
+                osVersion = lsb_release.Replace(Environment.NewLine, " ");
+                return;
+            }
+        }
+        catch(Exception ex)
+        {
```

Comment:
Nothing - on Redhat based Linux distros there might not be `lsb_release` so we ignore the error and try `/etc/redhat-release`